apps/src/code-studio/components/progress/UnitCalendar.jsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import UnitCalendarLessonChunk from './UnitCalendarLessonChunk';
+import {unitCalendarLessonChunk} from '@cdo/apps/templates/progress/unitCalendarLessonShapes';
+
+const weekWidth = 592;
+const styles = {
+  weekColumn: {
+    minWidth: 100,
+    backgroundColor: 'rgb(118, 101, 160)',
+    color: 'white',
+    textAlign: 'center',
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    fontWeight: 'bold',
+    minHeight: 50
+  },
+  scheduleColumn: {
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    width: weekWidth,
+    display: 'flex',
+    minHeight: 50
+  },
+  table: {
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    width: '100%'
+  },
+  key: {
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    width: '100%',
+    marginTop: 20
+  },
+  keyIcon: {
+    marginRight: 5
+  },
+  keySection: {
+    display: 'flex',
+    justifyContent: 'space-around',
+    width: '90%',
+    alignItems: 'center',
+    fontSize: 15
+  }
+};
+
+export default class UnitCalendar extends React.Component {
+  static propTypes = {
+    schedule: PropTypes.arrayOf(PropTypes.arrayOf(unitCalendarLessonChunk))
```

Comment:
It might seem silly but it could be helpful to define `PropTypes.arrayOf(unitCalendarLessonChunk)` as a shape in the shapes file called unitCalendarWeek.

Reply:
we actually no longer have unitCalendarWeeks anywhere once we move the scheduling logic to the UnitCalendar component